src/metabase/search/filter.clj

```diff
@@ -128,12 +128,15 @@
                            [:= :moderation_review.most_recent true]))
     (sql.helpers/where query false-clause)))
 
-(defmethod build-optional-filter-query [:verified "collection"]
+(defmethod build-optional-filter-query [:verified "card"]
   [_filter model query verified]
-  (assert (true? verified) "filter for non-verified collections is not supported")
-  (if (premium-features/has-feature? :official-collections)
-    (sql.helpers/where query [:= (search.config/column-with-model-alias model :authority_level) "official"])
-    (sql.helpers/where query false-clause)))
+  (assert (true? verified) "filter for non-verified cards is not supported")
+  (default-verified-card-query model query verified))
+
+(defmethod build-optional-filter-query [:verified "dataset"]
+  [_filter model query verified]
+  (assert (true? verified) "filter for non-verified cards is not supported")
+  (default-verified-card-query model query verified))
```

Comment:
I think you can just do
```suggestion
  (build-optional-filter-query filter "card" query verified)
```
And you wouldn't need a separate `default-verified-card-query` function.

The same tactic is used here
https://github.com/metabase/metabase/blob/74d9dc0df51ac3e4a6266952545cae43bbbeac92/src/metabase/search/config.clj#L123-L125